  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers --------------------------------------
  right_knuckle_position_controller:
    type: effort_controllers/JointPositionController
    joint: right_knuckle
    pid: {p: 100.0, i: 0.01, d: 10.0}
  left_knuckle_position_controller:
    type: effort_controllers/JointPositionController
    joint: left_knuckle
    pid: {p: 100.0, i: 0.01, d: 10.0}
  front_right_wheel_joint_position_controller:
    type: effort_controllers/JointVelocityController
    joint: front_right_wheel_joint
    pid: {p: 100.0, i: 0.01, d: 10.0, i_clamp: 100.0}
  front_left_wheel_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: front_left_wheel_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  Slider_position_controller:
    type: effort_controllers/JointPositionController
    joint: Slider
    pid: {p: 200.0, i: 0.10, d: 1.0}
  arm_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: arm_joint
    pid: {p: 100.0, i: 0.01, d: 50.0}
  gripper1_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: gripper1_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  gripper2_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: gripper2_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  tipper_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: tipper_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  lidar_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: lidar_joint
    pid: {p: 100.0, i: 0.5, d: 0.0}
